lib/bonuses/BonusEnum.h

```diff
@@ -12,6 +12,8 @@
 
 VCMI_LIB_NAMESPACE_BEGIN
 
+#include "../JsonNode.h"
```

Comment:
Forward-declaration should be enough I think?

Reply:
No, it is not, because we return a JsonNode.

Reply:
No, it is.
You can use forward-declared types as return type or as argument of a function declaration.

For function definition in cpp you will need full type definition though.

Reply:
Ok, done